BeVolt/App/src/CLI.c

```diff
@@ -0,0 +1,723 @@
+		} else {
+			printf("Error light is Off\n\r");
+		}
+		return;
+	}
+	switch(hashTokens[1]) {
+		case TEST:
+			printf("\n\r");
+			for(int i = 0; i < 10; i++) {
+				LED_Toggle(FAULT);
+				DelayMs(100);
+				LED_Toggle(RUN);
+				DelayMs(100);
+				LED_Toggle(UVOLT);
+				DelayMs(100);
+				LED_Toggle(OVOLT);
+				DelayMs(100);
+				LED_Toggle(OTEMP);
+				DelayMs(100);
+				LED_Toggle(OCURR);
+				DelayMs(100);
+				LED_Toggle(WDOG);
+				DelayMs(100);
+				LED_Toggle(CAN);
+				DelayMs(100);
+				LED_Toggle(EXTRA);
+				DelayMs(100);
+			}
+			break;
+		case FAULT_HASH:
+			toggleLED(FAULT);
+			break;
+		case RUN_HASH:
+			toggleLED(RUN);
+			break;
+		case OCURR_HASH:
+			toggleLED(OCURR);
+			break;
+		case OTEMP_HASH:
+			toggleLED(OTEMP);
+			break;
+		case OVOLT_HASH:
+			toggleLED(OVOLT);
+			break;
+		case WDOG_HASH:
+			toggleLED(WDOG);
+			break;
+		case CAN_HASH:
+			toggleLED(CAN);
+			break;
+		case EXTRA_HASH:
+			toggleLED(EXTRA);
+			break;
+		default:
+			printf("Invalid LED command\n\r");
+			break;
+	}
+}
+
+/** CLI_CAN
+ * Interacts with CAN by
+ * reading and writing to bus
+ */
+void CLI_CAN(void) {
+	uint8_t readData;
+	uint8_t writeData[8];
+	for(uint8_t i = 0; i < 7; i+= 2) {
+		char tempData[2] = {hexString[i], hexString[i+1]};
+		writeData[i/2] = strtol(tempData, NULL, 16);
+	}
+	switch(hashTokens[1]) {
+		case READ: 
+			while(CAN1_Read(&readData)){
+				printf("%d\n\r ", readData);
+			}
+		break;
+		case WRITE: 
+			switch(hashTokens[2]) {
+				case CAN_TRIP:
+					CAN1_Write(CAN_ID_BPS_TRIP, writeData, (strlen(hexString)/2)+(strlen(hexString)%2));
+					break;
+				case CLEAR:
+					CAN1_Write(CAN_ID_BPS_ALL_CLEAR, writeData, (strlen(hexString)/2)+(strlen(hexString)%2));
+					break;
+				case OFF:
+					CAN1_Write(CAN_ID_BPS_OFF, writeData, (strlen(hexString)/2)+(strlen(hexString)%2));
+					break;
+				case CURRENT:
+					CAN1_Write(CAN_ID_CURRENT_DATA, writeData, (strlen(hexString)/2)+(strlen(hexString)%2));
+					break;
+				case VOLTAGE:
+					CAN1_Write(CAN_ID_TOTAL_VOLTAGE_DATA, writeData, (strlen(hexString)/2)+(strlen(hexString)%2));
+					break;
+				case TEMPERATURE:
+					CAN1_Write(CAN_ID_AVG_TEMPERATURE_DATA, writeData, (strlen(hexString)/2)+(strlen(hexString)%2));
+					break;
+				case CHARGE:
+					CAN1_Write(CAN_ID_SOC_DATA, writeData, (strlen(hexString)/2)+(strlen(hexString)%2));
+					break;
+				case WATCHDOG:
+					CAN1_Write(CAN_ID_WDOG_TRIGGERED, writeData, (strlen(hexString)/2)+(strlen(hexString)%2));
+					break;
+				case ERROR:
+					CAN1_Write(CAN_ID_ERROR, writeData, (strlen(hexString)/2)+(strlen(hexString)%2));
+					break;
+				default:
+					printf("Invalid ID\n\r");
+					break;
+			}
+			break;
+		default:
+			printf("Invalid CAN command\n\r");
+			break;
+	}
+}
+	
+// TODO: Confirm status of Display;
+//		 if operational, display proper information (specifics found on Drive)
+/** CLI_Display
+ * Interacts with the display
+ */
+void CLI_Display(void) {}

```

Comment:
Might be a good idea to spew some error code to the CLI if this is called before we have a chance to implement it.

Reply:
Added a printf statement